report_artifact.go

```diff
@@ -216,24 +228,30 @@ func (a *TestArtifact) jsonToFTDC(ctx context.Context, path string) (string, err
 func (a *TestArtifact) gzip(path string) (string, error) {
 	srcFile, err := os.Open(path)
 	if err != nil {
-		return path, errors.Wrapf(err, "problem opening bson input file '%s'", path)
+		return path, errors.Wrapf(err, "opening BSON input file '%s'", path)
 	}
-	defer srcFile.Close()
+	defer func() {
+		_ = srcFile.Close()
+	}()
 
 	path += ".gz"
-	catcher := grip.NewCatcher()
 	outFile, err := os.Create(path)
 	if err != nil {
-		return path, errors.Wrapf(err, "problem opening ftdc output file '%s'", path)
+		return path, errors.Wrapf(err, "opening FTDC output file '%s'", path)
 	}
-	defer func() { catcher.Add(outFile.Close()) }()
+	defer func() {
+		_ = outFile.Close()
+	}()
 
+	catcher := grip.NewCatcher()
 	writer, err := gzip.NewWriterLevel(outFile, gzip.BestCompression)
 	if err != nil {
 		catcher.Add(err)
 		return path, catcher.Resolve()
 	}
-	defer func() { catcher.Add(writer.Close()) }()
+	defer func() {
+		_ = writer.Close()
+	}()
 
 	_, err = io.Copy(writer, srcFile)
 	catcher.Add(err)
```

Comment:
I would wrap this.

Reply:
Done. I had to change the logic a bit to wrap it, because the output file name overwrote the input file name parameter.